vllm/v1/metrics/loggers.py

```diff
@@ -203,8 +205,28 @@ def __init__(self,
         self.gauge_scheduler_waiting = make_per_engine(gauge_scheduler_waiting,
                                                        engine_indexes,
                                                        model_name)
+        if envs.VLLM_SERVER_DEV_MODE:
+            # Engine sleep state
+            self.gauge_engine_sleep_state = self._gauge_cls(
+                name="vllm:engine_sleep_state",
+                documentation=("Engine sleep state."),
+                labelnames=labelnames + ["sleep_state"],
+                multiprocess_mode="mostrecent")
+
+            # Setting default values
+            for idx in engine_indexes:
+                self.gauge_engine_sleep_state.labels(
```

Comment:
You should keep the metric objects returned by `labels()` around. Here's a pattern to copy:

```
        for reason in FinishReason:
	    self.counter_request_success[reason] = {
                idx:
		counter_request_success_base.labels(model_name, str(idx),
                                                    str(reason))
                for idx in engine_indexes
            }
```

Reply:
Following this pattern will also mean you don't need the model name in `record_sleep_state()`